File: MyTextField.rb
include OSX

class MyTextField < NSTextField
  # def initWithFrame frame
  #   super_initWithFrame frame
  #   tracking = NSTrackingArea.alloc.initWithRect_options_owner_userInfo [0, 0, frame[2], frame[3]],
  #                 NSTrackingMouseEnteredAndExited|NSTrackingInVisibleRect|NSTrackingActiveInKeyWindow, self, nil
  #   addTrackingArea tracking
  #   self
  # end
  attr_accessor :action2
  
  def mouseUp event
    p = convertPoint_fromView(event.locationInWindow, nil)
    if p.x < 30 and action2
      sendAction_to action2, target
    else
      sendAction_to action, target
    end
  end
  
  # def mouseEntered event
  #   setTextColor NSColor.redColor
  # end
  # 
  # def mouseExited event
  #   setTextColor NSColor.blackColor
  # end
end

File: MyTextField.m
//
//  MyTextField.m
//  ManiaDB
//
//  Created by Appledelhi on 11/27/08.
//  Copyright 2008 Appledelhi. All rights reserved.
//

#import "MyTextField.h"


@implementation MyTextField

- (void)mouseUp:(NSEvent*)event
{
  [super mouseUp:event];
  NSPoint p = [self convertPoint:[event locationInWindow] fromView:nil];
  _nameClicked = (p.x > 30);
  [self sendAction:[self action] to:[self target]];
}

- (BOOL)nameClicked
{
  return _nameClicked;
}

@end
